src/aria/widgets/container/Dialog.js

```diff
@@ -682,8 +848,23 @@ module.exports = Aria.classDefinition({
          */
         close : function () {
             var cfg = this._cfg;
-            if (this._popup) {
+            var modal = cfg.modal;
+
+            if (modal && cfg.waiAria) {
```

Comment:
To better respect the DRY principle, instead of checking again the same condition as in the `open` method (with the risk that the condition differs after changing the code only in one place), I think it is better to only check here whether `this._hiddenElements` is set (and make sure it is set only in the `open` method with the right condition).